src/generic_ui/scripts/ui.ts

```diff
@@ -190,11 +191,27 @@ module UI {
      * Upon construction, the UI installs update handlers on core.
      */
     constructor(
-        public core   :uproxy_core_api.CoreApi,
+        public core   :CoreConnector,
         public browserApi :BrowserAPI) {
       // TODO: Determine the best way to describe view transitions.
       this.view = ui_constants.View.SPLASH;  // Begin at the splash intro.
 
+      core.on('connect', () => {
```

Comment:
The problem is that connect and disconnect are too vague.  You can connect to a proxy (we usually call start), or you can connect to a social network (we usually call it login).  Any additional words we can add here will help explain what this actually does.